src/simulator/shape.py

```diff
@@ -13,3 +14,21 @@ def bounding_box(self, position: Position):
 
     def dump_info(self):
         pass
+
+    def rotation(x: float, y: float, z: float, phi: float, theta: float) -> list[float]:
+        """rotate the point by angles
+        Args:
+            x,y,z : coordinate of point in Cartesian
+            phi : angle of rotation about z-axes respectively
+            theta : angle of rotation about y-axes respectively
+        Return:
+            list of new coordinate point after rotation
+        """
+        cos_theta = cos(deg2rad(theta))
+        cos_phi = cos(deg2rad(phi))
+        sin_theta = sin(deg2rad(theta))
+        sin_phi = sin(deg2rad(phi))
+        new_x = round(x * cos_phi*cos_theta - y * sin_phi + z * sin_theta * cos_phi, 3)
```

Comment:
why do you round()?

Reply:
the sin and cos of angle(phi and theta) is a long number at the end :) , and it's not complete, for example sin(pi) isn't zero. it's 1.2246467991473532e-16 . so in testcase it was difficult for me , for that I use round() here.
if it's not necessary, I'll change them.

Reply:
yeah I know. but this must not be handled inside this function. In general, in computer arithmetics, whenever you want to have
if a == b
you write it as
if abs(a-b) < epsilon
or similar approaches.
In this function you don’t have such an if. so don’t round anything. In testcases you’ll run into that problem. But there we already have this included.